Gems/PhysX/Code/Source/PhysXCharacters/API/CharacterUtils.cpp

```diff
@@ -344,6 +344,55 @@ namespace PhysX
                 bool isAcceleration = true;
                 return physx::PxD6JointDrive(stiffness, damping, forceLimit, isAcceleration);
             }
+
+            AZStd::vector<DepthData> ComputeHierarchyDepths(const AZStd::vector<size_t>& parentIndices)
+            {
+                const size_t numNodes = parentIndices.size();
+                AZStd::vector<DepthData> nodeDepths(numNodes);
+                for (size_t nodeIndex = 0; nodeIndex < numNodes; nodeIndex++)
+                {
+                    nodeDepths[nodeIndex] = { -1, nodeIndex };
+                }
+
+                for (size_t nodeIndex = 0; nodeIndex < numNodes; nodeIndex++)
+                {
+                    if (nodeDepths[nodeIndex].m_depth != -1)
+                    {
+                        continue;
+                    }
+                    int depth = -1; // initial depth value for this node
+                    int ancestorDepth = 0; // the depth of the first ancestor we find when iteratively visiting parents
+                    bool ancestorFound = false; // whether we have found either an ancestor which already has a depth value, or the root
+                    size_t currentIndex = nodeIndex;
+                    while (!ancestorFound)
+                    {
+                        depth++;
+                        if (depth > numNodes)
+                        {
+                            AZ_Assert(false, "Loop detected in ragdoll configuration.");
+                            return nodeDepths;
+                        }
```

Comment:
If this is an assert, meaning it's a programmer error, then it should never happen and therefore there is no need to have code making it safe.

All this can be replaced by:
````
AZ_Assert(depth <= numNodes, "Loop detected in ragdoll configuration.");
````

If there is some `parentIndices` passed to this function that could make this happen, then keep it as it is and change it for `AZ_Error`.

Reply:
Yeah, it's more of a data error rather than a programming error, so I'll change it to an AZ_Error.